macrosynergy/dataquery/api.py

```diff
@@ -61,22 +60,29 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
         if exc_type:
             print(f'Execution {exc_type} with value (exc_value):\n{exc_value}')
 
+    def check_access(self):
+        """
+        Method to validate the Class used to access DataQuery. The access Class will be
+        controlled by the oauth parameter.
+        """
+
+        return self.access
+
     def check_connection(self) -> bool:
-        """Check connect (heartbeat) to DataQuery"""
-        js: dict = self.access.get_dq_api_result(url=self.access.base_url + "/services/heartbeat")
+        """
+        Check connection (heartbeat) to DataQuery.
+        """
+        endpoint = "/services/heartbeat"
+        js: dict = self.access.get_dq_api_result(url=self.access.base_url + endpoint)
 
         results: dict = js["info"]
-        if int(results["code"]) != 200:
-            print(
-                f"DataQuery response {results['message']:s}"
-                f" with description: {results['description']:s}"
-            )
```

Comment:
If int(results["code"]) is not equal to 200, the above is printed. 

Happy to include both if necessary but thought the latter would also work.

Reply:
Add `logging` for it, so we can at least easily see the message with a logger if needed.

To do so, add at the start of the file:
```python
import logging
logger = logging.getLogger(__name__)
```

and in the `check_connection` function:
```python
if int(results["code"] != 200:
    logger.error("DataQuery response %s with description: %s", results["message"], results["description"])
```

This will make it easier to debug any issues (in general we should have more logging statements in our code).

Reply:
Added.